sound/soc/bcm/adau1977-adc.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * ASoC Driver for ADAU1977 ADC
+ *
+ * Author:	Andrey Grodzovsky <andrey2805@gmail.com>
+ *		Copyright 2016
+ *
+ * This file is based on hifibery_dac driver by Florian Meier.
+ * 
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ */
+
+#include <linux/module.h>
+#include <linux/platform_device.h>
+
+#include <sound/core.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/jack.h>
+
+enum adau1977_clk_id {
+    ADAU1977_SYSCLK,
+};
+ 
+enum adau1977_sysclk_src {
+    ADAU1977_SYSCLK_SRC_MCLK,
+    ADAU1977_SYSCLK_SRC_LRCLK,
+};
+
+static int eval_adau1977_init(struct snd_soc_pcm_runtime *rtd)
+{
+	int ret;
+	struct snd_soc_dai *codec_dai = rtd->codec_dai;
+	
+	ret = snd_soc_dai_set_tdm_slot(codec_dai, 0, 0, 0, 0);
+	if (ret < 0)
+		return ret;
+
+	return snd_soc_codec_set_sysclk(rtd->codec, ADAU1977_SYSCLK,
+			ADAU1977_SYSCLK_SRC_MCLK, 11289600, SND_SOC_CLOCK_IN);
+}
+ 
+static struct snd_soc_dai_link snd_rpi_adau1977_dai[] = {
+	{
+	.name = "adau1977",
+	.stream_name = "ADAU1977", 
+	.cpu_dai_name = "bcm2708-i2s.0", 
+	.codec_dai_name = "adau1977-hifi",
+	.platform_name = "bcm2708-i2s.0",
+	.codec_name = "adau1977.1-0011",
+	.init = eval_adau1977_init,
+	.dai_fmt = SND_SOC_DAIFMT_I2S |
+		SND_SOC_DAIFMT_NB_NF |
+		SND_SOC_DAIFMT_CBM_CFM,
+	},
+};
+
+/* audio machine driver */
+static struct snd_soc_card snd_adau1977_adc = {
+	.name         = "snd_rpi_adau1977_adc",
+	.owner        = THIS_MODULE,
+	.dai_link     = snd_rpi_adau1977_dai,
+	.num_links    = ARRAY_SIZE(snd_rpi_adau1977_dai),
+};
+
+static int snd_adau1977_adc_probe(struct platform_device *pdev)
+{
+	int ret = 0;
+
+	snd_adau1977_adc.dev = &pdev->dev;
+	dev_info(&pdev->dev,"%s:%d\n",__func__,__LINE__);
```

Comment:
Remove this line - it doesn't tell you anything useful. You will see ALSA messages for success and error.